packages/open-payments/src/client/grant.ts

```diff
@@ -0,0 +1,58 @@
+import { HttpMethod } from 'openapi'
+import { ClientDeps } from '.'
+import {
+  getASPath,
+  InteractiveGrant,
+  InteractiveGrantRequest,
+  NonInteractiveGrant,
+  NonInteractiveGrantRequest
+} from '../types'
+import { post } from './requests'
+
+interface RequestGrantArgs<T> {
+  url: string
+  request: T
+}
+
+export interface GrantRoutes {
+  requestInteractiveGrant(
+    args: RequestGrantArgs<InteractiveGrantRequest>
+  ): Promise<InteractiveGrant>
+  requestNonInteractiveGrant(
+    args: RequestGrantArgs<NonInteractiveGrantRequest>
+  ): Promise<NonInteractiveGrant>
+}
+
+export const createGrantRoutes = (clientDeps: ClientDeps): GrantRoutes => {
+  const { axiosInstance, openApi, logger } = clientDeps
+
+  const createInteractiveGrantValidator =
+    openApi.createResponseValidator<InteractiveGrant>({
+      path: getASPath('/'),
+      method: HttpMethod.POST
+    })
+  const createNonInteractiveGrantValidator =
+    openApi.createResponseValidator<NonInteractiveGrant>({
+      path: getASPath('/'),
+      method: HttpMethod.POST
+    })
```

Comment:
unfourtunately, needed both to pass type checking when given as arguments to the `request*Grant` methods

Reply:
If we keep `requestInteractiveGrant`/`requestNonInteractiveGrant` separate, I think we'll need custom validators that include the `openApi.createResponseValidator` as well as a subsequent type guard that the response is the expected `InteractiveGrant` or `NonInteractiveGrant`. (In which case, we might be able to have a single `openApi.createResponseValidator` for both?)
As is, I think the OpenAPI validator would allow a `NonInteractiveGrant` response for `requestInteractiveGrant` and vice versa.